staging/src/github.com/labring/image-cri-shim/pkg/shim/config.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2022 cuisongliu@qq.com.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package shim
+
+import (
+	"encoding/base64"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/labring/image-cri-shim/pkg/cri"
+	"github.com/labring/image-cri-shim/pkg/server"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	api "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+	"sigs.k8s.io/yaml"
+
+	fileutil "github.com/labring/sealos/pkg/utils/file"
+	img "github.com/labring/sealos/pkg/utils/images"
+	"github.com/labring/sealos/pkg/utils/logger"
+)
+
+type Config struct {
+	ShimSocket  string                    `json:"shim"`
+	ImageSocket string                    `json:"cri"`
+	Address     string                    `json:"address"`
+	Force       bool                      `json:"force"`
+	Debug       bool                      `json:"debug"`
+	Image       string                    `json:"image"`
+	Sync        int64                     `json:"sync"`
+	Auth        string                    `json:"auth"`
+	Base64Auth  string                    `json:"-"`
+	Username    string                    `json:"-"`
+	Password    string                    `json:"-"`
+	Domain      string                    `json:"-"`
+	CRIConfigs  map[string]api.AuthConfig `json:"-"`
+}
+
+func (c *Config) PreProcess() error {
+	if c.ShimSocket == "" {
+		c.ShimSocket = server.SealosShimSock
+	}
+	defer func() {
+		if c.Image != "" {
+			if c.Sync != 0 {
+				go wait.Forever(func() {
+					images, err := img.LoadImages(c.Image)
+					if err != nil {
+						logger.Warn("load images from image dir: %v", err)
+					}
+					server.ShimImages.Insert(images...)
+					logger.Info("sync image list for image dir,sync second is %d,data is %+v", c.Sync, images)
+				}, time.Duration(c.Sync*int64(time.Second)))
+			} else {
+				//cache
+				images, err := img.LoadImages(c.Image)
+				if err != nil {
+					logger.Warn("load images from image dir: %v", err)
+				}
+				server.ShimImages.Insert(images...)
+				logger.Info("sync image list for image dir,sync once second is %d,data is %+v", c.Sync, images)
+			}
+		}
+	}()
+	logger.Info("shim-socket: %s", c.ShimSocket)
```

Comment:
print log in one line

Reply:
this this print shim socket. after this print image socket